core/src/test/scala/unit/kafka/server/MetadataRequestTest.scala

```diff
@@ -35,6 +37,43 @@ class MetadataRequestTest extends BaseRequestTest {
   }
 
   @Test
+  def testClusterIdWithAllTopics() {
+    val v2MetadataResponse = sendMetadataRequest(MetadataRequest.allTopics(), 2)
+    val v2ClusterId = v2MetadataResponse.clusterId()
+    assertNotNull("clusterId should not be null", v2ClusterId)
+
+    val v1MetadataResponse = sendMetadataRequest(MetadataRequest.allTopics(), 1)
+    val v1ClusterId = v1MetadataResponse.clusterId()
+    assertEquals(s"v1 clusterId should be ${MetadataResponse.NO_CLUSTER_ID}", v1ClusterId, MetadataResponse.NO_CLUSTER_ID)
+  }
+
+  @Test
+  def testClusterIdWithNoTopics() {
```

Comment:
I think this should be named `testClusterIdWithRequestVersion1` and we don't need `testClusterIdWithAllTopics`.